opensrp-eusm/src/main/java/org/smartregister/eusm/repository/AppLocationRepository.java

```diff
@@ -71,7 +71,7 @@ public Set<Location> getLocationByNameAndGeoLevel(Set<String> names, String leve
         if (names == null || names.isEmpty())
             return locations;
         try (Cursor cursor = getReadableDatabase().rawQuery("SELECT * FROM " + getLocationTableName() +
-                " WHERE " + StringUtils.repeat(NAME + " =?", " OR ", names.size()) + " AND " + GEOGRAPHICAL_LEVEL + "=?", ArrayUtils.add(names.toArray(new String[0]), level))) {
+                " WHERE " + StringUtils.repeat("(" + NAME + " =? AND " + GEOGRAPHICAL_LEVEL + " = " + level + ")", " OR ", names.size()), names.toArray(new String[0]))) {
```

Comment:
```suggestion
                " WHERE (" + StringUtils.repeat(NAME + " =?", " OR ", names.size()) + ") AND " + GEOGRAPHICAL_LEVEL + "=?", ArrayUtils.add(names.toArray(new String[0]), level))) {
```
The previous query can still work by surrounding it with parenthesis.

Reply:
The above should achieve the existing parameter binding.